server/init.lua

```diff
@@ -0,0 +1,26 @@
+-- Import section
+
+Event = require("event")
+
+local addressesConfigFile = "configuration.addresses"
+local addresses = require(addressesConfigFile)
+local addMachine = require("usecases.add-machine")
+local getMultiblockStatus = require("usecases.get-multiblock-status")
+
+--
+
+local function updateMachineList(_, address, _)
+    local comp = Component.proxy(address)
+    if comp.type == "waypoint" or comp.type == "gt_machine" or comp.type == "gt_batterybuffer" then
+        addMachine(address, addressesConfigFile)
+    end
+end
+Event.listen("component_added", updateMachineList)
+
+local statuses = {}
+while true do
```

Comment:
Is this an infinite loop by design? Is there a better way to do this?

Reply:
It's the main server loop.
Everything the server does should stem from this file

Reply:
We should have a main loop somewhere. Not in the server file